shared/slices/slice_generic.go

```diff
@@ -0,0 +1,308 @@
+package slices
+
+import (
+	"fmt"
+	"reflect"
+)
+
+func interfaceToSlice(slice interface{}) []interface{} {
+	s := reflect.ValueOf(slice)
+	if s.Kind() != reflect.Slice {
+		panic("InterfaceSlice() given a non-slice type")
+	}
+	ret := make([]interface{}, s.Len())
+	for i := 0; i < s.Len(); i++ {
+		ret[i] = s.Index(i).Interface()
+	}
+	return ret
+}
+
+// GenericIntersection returns a new set with elements that are common in
+// both sets a and b.
+func GenericIntersection(a, b interface{}) (reflect.Value, error) {
+
+	switch v := a.(type) {
+	case []uint32:
+
+		set := make([]uint32, 0)
```

Comment:
This whole chunk of code is repeated three times 
```go
set := make([]uint32, 0)
		sliceType := reflect.TypeOf(set)
		modifiedSet := reflect.MakeSlice(sliceType, 0, 0)
		m := make(map[uint32]bool)

		set1 := interfaceToSlice(a)
		set2 := interfaceToSlice(b)
		for i := 0; i < len(set1); i++ {
			m[set1[i].(uint32)] = true
		}
```
This can be easily abstracted out, the same applies for all the other types.

Reply:
will remove redundant code